src/explore-education-statistics-frontend/src/modules/table-tool/components/TableToolShare.tsx

```diff
@@ -57,54 +64,55 @@ const TableToolShare = ({
     ) as HTMLInputElement;
     el?.select();
     document.execCommand('copy');
+    setScreenReaderMessage('Link copied to the clipboard.');
   };
 
-  if (!permalinkId) {
-    return (
-      <>
-        <h3 className="govuk-heading-s">Save table</h3>
-        <LoadingSpinner
-          alert
-          inline
-          loading={permalinkLoading}
-          size="sm"
-          text="Generating permanent link"
-        >
-          <ButtonText onClick={handlePermalinkClick}>
-            Generate shareable link
-          </ButtonText>
-        </LoadingSpinner>
-      </>
-    );
-  }
-
   return (
-    <div className="dfe-align--left">
-      <h3 className="govuk-heading-s">Generated share link</h3>
-
-      <div className="govuk-inset-text">
-        Use the link below to see a version of this page that you can bookmark
-        for future reference, or copy the link to send on to somebody else to
-        view.
-      </div>
-
-      <p className="govuk-!-margin-top-0 govuk-!-margin-bottom-2">
-        <UrlContainer
-          data-testid="permalink-generated-url"
-          url={`${window.location.origin}/data-tables/permalink/${permalinkId}`}
-        />
-      </p>
-
-      <ButtonGroup>
-        <Button variant="secondary" onClick={handleCopyClick}>
-          Copy link
-        </Button>
-
-        <ButtonLink to={`/data-tables/permalink/${permalinkId}`}>
-          View share link
-        </ButtonLink>
-      </ButtonGroup>
-    </div>
+    <>
+      {!permalinkId ? (
```

Comment:
Why use a ternary? It seemed cleaner previously with the early return, plus we've ended up making loads of changes to the markup as a result

Reply:
It's because the screen reader message div needs to always be present for the message to be announced as it's looking for changes in the aria-live area.

Reply:
Oh I see! I thought the screen reader message was in the lower branch of the ternary! My bad :sweat_smile: